projects/s-rxjs-utils/src/lib/map-with-caching.ts

```diff
@@ -0,0 +1,46 @@
+import { ObjectWith } from "micro-dash";
+import { map } from "rxjs/operators";
+
+/**
+ * Applies a given function to each value in the array emitted by the source Observable, and emits the resulting array as an Observable. Each item in the array is cached using key generated by the given function so that the resulting arrays contain references to the exact same objects included in the previous emission, when possible.
```

Comment:
`[yellow]`
```suggestion
 * Applies a given function to each item in the upstream array and emits the result. Each item is then cached using the key generated by `buildCacheKey` so that the next emission contains references to the matching objects from previous emission, without running `buildDownstreamItem` again. The cache is only held between successive emissions.
```

It would also be nice to also include a sentence "This is useful e.g. when ..."